### DISCUSSION TITLE
understanding fine grained admin permission - manage user group

### DISCUSSION THREAD

**Discussion by manuschillerdev** [keycloak/keycloak#13749] [Q&A]
I am trying to configure following scenario in Keycloak (bitnami/keycloak:18)

Keycloak 
```
-- realms
---- master -> sys-admin
---- project-realm
------ client app: ui
------ user groups
-------- org1-user-admin
-------- org1-ui-user
-------- org2-user-admin
-------- org2-ui-user
...
-------- orgN-user-admin
-------- orgN-ui-user
```

**Roles:**
* sys-admin: may manage all realms (out of scope for the question)
* orgN-user-admin: 
  * should be able to manage and see only users in orgN-ui-user
  * can not see users of other user groups than orgN-ui-user
  * can not administrate other resources (clients, roles, identity providers, ... anything) than exactly the users in orgN-ui-user
* orgN-ui-user: 
  * can not self-register - can only be managed by orgN-user-admin
  * should only be able to see the account console and manage its own settings (locale, name, and so on)
  * can login to the client app ui via Authorization Code Flow

**Questions**
1) Is this possible with fine-grained admin permissions?
2) how could a configuration look like for that scenario?

**Comment by ssilvert** [comment]
That all looks doable to me.  I don't have time to walk you through it, but you can post follow up discussions if you can't find the answers you are looking for.

**Reply by ichbindannmalwegmann** [reply]
It is possible to restrict the list of users a "admin" sees to a group, but to edit a user (no matter in which group the a user is), a admin **must** have the permission "view-users" (what is more the role "view-users"). But if an admin has the role "view-users", he sees and can edit all users - which is not what we want. So I would be interested in how I can configure the system so that a certain admin can only administrate (see and edit) a certain group of users.

**Comment by ichbindannmalwegmann** [comment]
I planned to do the same but I couldn't find out how to manage that orgX-user-admin can only see and manage orgX users. Did you found out how to do it? If I assign a user/group to role "view-users" it always sees all users, no matter if I add policies that should restrict the permission to the groups.

**Reply by manuschillerdev** [reply]
Unfortunately no. Would you mind sharing your config with me? Maybe I can elaborate on your solution :)

**Reply by ichbindannmalwegmann** [reply]
At the moment I don't have something that is worth to play around with.

**Comment by ssilvert** [comment]
I haven't gone through this thread to see if it applies, but just be aware that there is a fix for fine grained permission for the new admin console.  This fix will be in 19.0.2.

So you might want to check to see if what you are wanting to do works in the old admin console.
